lib/ansible/executor/process/worker.py

```diff
@@ -125,6 +113,51 @@ def _hard_exit(self, e):
 
         os._exit(1)
 
+    @contextlib.contextmanager
+    def _detach(self):
+        '''
+        The intent here is to detach the child process from the inherited stdio fds,
+        including /dev/tty. Children should use Display instead of direct interactions
+        with stdio fds.
+        '''
+        try:
+            os.closerange(0, 3)
+            os.setsid()
+            sys.stdin = sys.__stdin__ = os.fdopen(  # type: ignore[misc]
+                os.dup2(
+                    os.open(os.devnull, os.O_RDWR),
+                    0,
+                ),
+                'r'
+            )
+            # Close stdin so we don't get hanging workers
+            # We use sys.stdin.close() instead of os.close(0) for places where sys.stdin is used,
+            # to give better errors
+            sys.stdin.close()
+            sys.stdout = sys.__stdout__ = os.fdopen(  # type: ignore[misc]
+                os.dup2(
+                    os.open(os.devnull, os.O_WRONLY),
+                    1
+                ),
+                'w'
+            )
```

Comment:
I tested that I can create a FD that goes to `/dev/null` for all external apps, but set `sys.stdout`/`sys.stderr` to `io.StringIO`.  This would be the best way to verify whether the python code of the worker tried to output.  Trying to make FD 1 or 2 a pipe has many complexities to overcome, that I don't want to actually pursue, as I don't think it gives us much.

So if we want to at least capture python attempts at writing to stdout/stderr `io.StringIO` is the easiest. Other than it may present issues if someone tries to send `sys.stdin`/`sys.stderr` through some API that requires `fileno()` to be implemented, but I think we are probably ok with that.  Although if it some dependency that does it, it could make it hard for plugin authors to make use of such a dep.